yt/frontends/gadget_fof/data_structures.py

```diff
@@ -45,7 +45,13 @@
 
 class GadgetFOFParticleIndex(ParticleIndex):
     def __init__(self, ds, dataset_type):
-        super(GadgetFOFParticleIndex, self).__init__(ds, dataset_type)
+        self.dataset_type = dataset_type
+        self.dataset = weakref.proxy(ds)
+        self.float_type = np.float64
+        self._setup_data_io()
+        self._setup_filenames()
+        super(ParticleIndex, self).__init__(ds, dataset_type)
```

Comment:
Can you explain why the change in the super call?  I'm not sure I follow.

Reply:
Oh yeah, this was so I could change the ordering of some things that happen in `__init__`. Specifically, I think I needed `_setup_data_io` and `_setup_filenames` to happen before things that happen in `Index.__init__`. Much of this change is taking what happens in `ParticleIndex.__init__` and moving it here. That's why I don't switched from calling `super(GadgetFOFParticleIndex, ...)` to `super(ParticleIndex, ...)`.